nilearn/regions/rena_clustering.py

```diff
@@ -171,7 +295,12 @@ def _weighted_connectivity_graph(X, mask_img):
     """
     n_features = X.shape[1]
 
-    edges, weight = _make_edges_and_weights(X, mask_img)
+    if isinstance(mask_img, SurfaceImage):
+        edges, weight = _make_edges_and_weights_surface(X, mask_img)
+    else:
+        edges, weight = _make_edges_and_weights(X, mask_img)
+
+    # TODO : deal with dict results if surface analysis
```

Comment:
Right now failure is here because edges and weights are returned in a dictionary with keys representing the surface image object mesh and data keys for the different hemispheres

Reply:
We could great a unique graph that would stack the data from the two hemispheres. This graph would have 2 connected components. This would have the advantage to make the use on surface transparent for other functions.